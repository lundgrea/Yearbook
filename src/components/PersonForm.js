import React, { Component } from 'react'

class PersonForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      quote: '',
      superlative: '',
      image: ''
    };
  }

  handleFormUpdate = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitPerson = (event) => {
    event.preventDefault();
    this.props.addNewPerson(this.state);
    this.setState({name: '', quote: '', superlative: '', image: ''})
  }

  render() {
    return (
      <form>
        <input
          type="text"
          placeholder="Name"
          name="name"
          value={this.state.name}
          onChange={event => this.handleFormUpdate(event)}
        />
         <input
          type="text"
          placeholder="Quote"
          name="quote"
          value={this.state.quote}
          onChange={event => this.handleFormUpdate(event)}
        />
        <input
          type="text"
          placeholder="Superative"
          name="superlative"
          value={this.state.superlative}
          onChange={event => this.handleFormUpdate(event)}
        />
        <button onClick={event => this.submitPerson(event)}> Add Person </button>
      </form>
    )
  }
}

export default PersonForm