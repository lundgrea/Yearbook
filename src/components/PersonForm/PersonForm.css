form {
  border: 2px solid #EBEBEB;
  margin: 50px 30px 20px 20px;
  padding: 20px 20px 20px 20px;
  color: #EBEBEB;
  border-radius: 10px;
}

::placeholder{
  font-size: 16px;
}

