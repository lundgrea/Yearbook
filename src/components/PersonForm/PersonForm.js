import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './PersonForm.css';


class PersonForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      quote: '',
      superlative: '',
      image: ''
    };
  }

  handleFormUpdate = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitPerson = (event) => {
    event.preventDefault();
    this.props.addNewPerson(this.state);
    this.setState({name: '', quote: '', superlative: '', image: ''})
  }

  render() {
    return (
      <form>
      <h3> Add New Student To The Yearbook</h3>
        <input
          type="text"
          placeholder="Name"
          name="name"
          value={this.state.name}
          onChange={event => this.handleFormUpdate(event)}
          className="input-name"
        />
         <input
          type="text"
          placeholder="Quote"
          name="quote"
          value={this.state.quote}
          onChange={event => this.handleFormUpdate(event)}
          className="input-quote"
        />
        <input
          type="text"
          placeholder="Superative"
          name="superlative"
          value={this.state.superlative}
          onChange={event => this.handleFormUpdate(event)}
          className="input-superlative"

        />
        <button className="add-button" onClick={event => this.submitPerson(event)}> Add Person </button>
      </form>
    )
  }
}

export default PersonForm

PersonForm.propTypes = {
addNewPerson: PropTypes.func
}