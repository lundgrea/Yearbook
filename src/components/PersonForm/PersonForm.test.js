import React from 'react';
import { shallow } from 'enzyme';
import PersonForm from './PersonForm';

describe('PersonForm', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<PersonForm 
        addNewPerson={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })


  it('should change the state on input within the form', () => {
    const wrapper = shallow(<PersonForm
      addNewPerson={jest.fn()}
    />);

    const input = wrapper.find('.input-name');
    input.simulate('change', { target: { value: 'George', name: 'name' } })
    expect(wrapper.state('name')).toEqual('George');
  })





  it('should fire the submitPerson method on the click of the submit button' , () => {

  })
})