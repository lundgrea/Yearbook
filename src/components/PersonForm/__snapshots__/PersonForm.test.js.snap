// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonForm should match the snapshot with all data passed in correctly 1`] = `
<form>
  <h3>
     Add New Student To The Yearbook
  </h3>
  <input
    className="input-name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="input-quote"
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <input
    className="input-superlative"
    name="superlative"
    onChange={[Function]}
    placeholder="Superative"
    type="text"
    value=""
  />
  <button
    className="add-button"
    onClick={[Function]}
  >
     Add Person 
  </button>
</form>
`;
